File: languages/java/runtime/src/main/java/co/revaly/sdk/errors/TransientFailureException.java
package co.revaly.sdk.errors;

import com.fasterxml.jackson.databind.JsonNode;

/**
 * The payment was definitively NOT processed at RAP-core: either the transport can prove the
 * request was never sent (connection refused, DNS or TLS failure before acceptance), or the
 * platform answered 503 with {@code code: not_processed} (provable non-dispatch). <b>Safe to fail
 * over</b> — route the payment to your own gateway immediately (failover-contract §2, the PRD's
 * first-layer failover path).
 */
public final class TransientFailureException extends RapCoreException {
    private static final long serialVersionUID = 1L;

    TransientFailureException(
            Integer statusCode,
            String code,
            String errorMessage,
            JsonNode details,
            String correlationId,
            String rawErrorBody) {
        super(
                RapFailureClass.TRANSIENT_FAILURE,
                buildMessage(
                        RapFailureClass.TRANSIENT_FAILURE,
                        statusCode,
                        code,
                        correlationId,
                        "definitively not processed; safe to fail over"),
                statusCode,
                code,
                errorMessage,
                details,
                correlationId,
                rawErrorBody,
                null);
    }

    TransientFailureException(String detail, Throwable cause) {
        super(
                RapFailureClass.TRANSIENT_FAILURE,
                buildMessage(RapFailureClass.TRANSIENT_FAILURE, null, null, null, detail),
                null,
                null,
                null,
                null,
                null,
                null,
                cause);
    }
}

File: languages/php/runtime/src/Errors/TransientFailureException.php
<?php

declare(strict_types=1);

namespace Revaly\Sdk\Errors;

/**
 * Definitively not processed — safe to fail over immediately (failover-contract §2).
 * Raised only on client-provable never-sent transport failures (DNS, TCP connect, TLS
 * handshake — proven by the transport's own phase semantics) and on `503` with
 * `code: not_processed` (the platform's provable non-dispatch signal, P-1).
 */
final class TransientFailureException extends RapCoreException
{
    public function getFailureClass(): RapFailureClass
    {
        return RapFailureClass::TransientFailure;
    }
}
